### PR TITLE
Story1task4

### PR DESCRIPTION
**Pull request by Joshsilas**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -6,13 +6,27 @@
 $query->execute();
 $boardgames = $query->fetchAll();
 
-function creategamecard($name, $playercount, $difficulty, $description){
+class collectioncardmaker {
```
Comment (alexjames4) [inline functions.php]: capital letter for object name.

also file should be called Collectioncardmaker.php

File: functions.php [hunk 2]
```diff
@@ -6,13 +6,27 @@
...
+    public string $name;
+    public string $playercount;
+    public string $difficulty;
+    public string $description;
+
+public function __construct($name, $playercount, $difficulty, $description)
+{
+    $this->name =$name;
```
Comment (alexjames4) [inline functions.php]: space after equals for all in the construct function

File: functions.php [hunk 3]
```diff
@@ -6,13 +6,27 @@
...
 $boardgames = $query->fetchAll();
 
-function creategamecard($name, $playercount, $difficulty, $description){
+class collectioncardmaker {
+    public string $name;
+    public string $playercount;
+    public string $difficulty;
+    public string $description;
+
+public function __construct($name, $playercount, $difficulty, $description)
```
Comment (alexjames4) [inline functions.php]: indentation

File: functions.php [hunk 4]
```diff
@@ -6,13 +6,27 @@
...
+{
+    $this->name =$name;
+    $this->playercount =$playercount;
+    $this->difficulty =$difficulty;
+    $this->description =$description;
+}
+public function creategamecard()
```
Comment (alexjames4) [inline functions.php]: indentation

File: functions.php [hunk 5]
```diff
@@ -6,13 +6,27 @@
...
     return "
     <section class = 'board-game-section'>		
-				<h2> $name </h2>
-				<p>The Player Count is $playercount players</p>
-				<p> The difficulty to learn is $difficulty/5</p>
-				<p>$description </p>
-			</section>	";
+		<h2> $this->name </h2>
+		<p>The Player Count is $this->playercount players</p>
+		<p> The difficulty to learn is $this->difficulty/5</p>
```
Comment (alexjames4) [inline functions.php]: unnecessary space after <p>

File: functions.php [hunk 6]
```diff
@@ -6,13 +6,27 @@
...
+		<p>$this->description </p>
+	</section>	";
```
Comment (alexjames4) [inline functions.php]: </section>	";

large gap between > and "

File: functions.php [hunk 7]
```diff
@@ -5,6 +5,28 @@
 $query = $db ->prepare("SELECT `name`, `description`, `playercount`, `difficulty` FROM `boardgames`");
```
Comment (alexjames4) [inline functions.php]: move all this stuff to the index

**Review by alexjames4 — CHANGES_REQUESTED** [review]
minor changes but basically there well done

File: collectioncardmaker.php [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+class CollectionCardMaker {
```
Comment (alexjames4) [inline collectioncardmaker.php]: naming of the php file should be the same as this: CollectionCardMaker.php

check that all the require_once() and links etc match this

File: collectioncardmaker.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+         $this->name = $name;
+         $this->playercount = $playercount;
+         $this->difficulty = $difficulty;
+         $this->description = $description;
+    }
+    public function createGameCard()
+    {
+     return 
+        "<section class = 'board-game-section'>		
+		    <h2> $this->name </h2>
```
Comment (alexjames4) [inline collectioncardmaker.php]: I think these spaces are not needed

File: collectioncardmaker.php [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+		    <p>The Player Count is $this->playercount players</p>
+		    <p> The difficulty to learn is $this->difficulty/5</p>
```
Comment (alexjames4) [inline collectioncardmaker.php]: <p> The

this too has a space

File: index.php [hunk 11]
```diff
@@ -15,8 +20,8 @@
 		<div class = "container">
 			<?php
 			foreach ($boardgames as $game){
-				$collectioncardmaker = new collectioncardmaker($game['name'], $game['playercount'], $game['difficulty'], $game['description']);
-				echo $collectioncardmaker->creategamecard();
+				$collectioncardmaker = new CollectionCardMaker($game['name'], $game['playercount'], $game['difficulty'], $game['description']);
```
Comment (alexjames4) [inline index.php]: $collectioncardmaker this should be camel case throughout

File: index.php [hunk 12]
```diff
@@ -15,8 +20,8 @@
 		<div class = "container">
 			<?php
 			foreach ($boardgames as $game){
```
Comment (alexjames4) [inline index.php]: would it be better to refer to it like: ($boardgames as $boardgame)?

File: index.php [hunk 13]
```diff
@@ -15,8 +20,8 @@
...
-				$collectioncardmaker = new collectioncardmaker($game['name'], $game['playercount'], $game['difficulty'], $game['description']);
-				echo $collectioncardmaker->creategamecard();
+				$collectioncardmaker = new CollectionCardMaker($game['name'], $game['playercount'], $game['difficulty'], $game['description']);
+				echo $collectioncardmaker->createGameCard();
```
Comment (alexjames4) [inline index.php]: camelcase for the variable

File: index.php [hunk 14]
```diff
@@ -1,6 +1,11 @@
 <?php 
-require_once('functions.php');
-
+require_once('CollectionCardMaker.php');
```
Comment (alexjames4) [inline index.php]: for the unit testing task you will need to put this in a src folder and then update all of the requires to match this

**Review by alexjames4 — CHANGES_REQUESTED** [review]
minor things again

**Review by CMeier556 — APPROVED** [review]

**Pull request merged**